pkg/operator/loglevel/controller.go

```diff
@@ -0,0 +1,128 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loglevel
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cloud-credential-operator/pkg/operator/status"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/utils"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	logLevelUtils "github.com/openshift/library-go/pkg/operator/loglevel"
+)
+
+const (
+	controllerName = "loglevel"
+)
+
+// Add creates a new ConfigController and adds it to the Manager.
+func Add(mgr manager.Manager, kubeConfig string) error {
+	r := newReconciler(mgr)
+
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to CloudCredential object and reconcile the loglevel changes
+	err = c.Watch(
+		&source.Kind{Type: &operatorv1.CloudCredential{}},
+		&handler.EnqueueRequestForObject{})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcileCloudCredentialConfig{
+		Client: mgr.GetClient(),
+	}
+	status.AddHandler(controllerName, r)
+
+	return r
+}
+
+var _ reconcile.Reconciler = &ReconcileCloudCredentialConfig{}
+
+// ReconcileCloudCredentialConfig reconciles a CredentialConfig object
+type ReconcileCloudCredentialConfig struct {
+	client.Client
+}
+
+// Reconcile reads the state of the cluster for a CredentialConfig object and
+// makes changes based on the state read and what is in the CredentialConfig.LogLevel
+// Automatically generate RBAC rules to allow the Controller to read and write required types.
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=cloudcredential/spec,verbs=get
+func (r *ReconcileCloudCredentialConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithFields(log.Fields{
+		"controller": controllerName,
+		"cr":         fmt.Sprintf("%s/%s", request.NamespacedName.Namespace, request.NamespacedName.Name),
+	})
+
+	// Check OperatorLogLevel and LogLevel
+	log.Debug("Checking log level for Reconcile request")
+	desiredOperatorLogLevel, err := utils.GetOperatorLogLevel(r, logger)
+	if err != nil || !logLevelUtils.ValidLogLevel(desiredOperatorLogLevel) {
+		return reconcile.Result{Requeue: false}, fmt.Errorf("the operatorLogLevel defined for the cloudcredential object is invalid")
+	}
```

Comment:
this can error out for two reasons
1) error fetching the desired log level on CredentialConfig
2) desired log level is invalid

I would go with the following
```suggestion
	desiredOperatorLogLevel, err := utils.GetOperatorLogLevel(r, logger)
	if err != nil {
		return reconcile.Result{}, err
	}
	if !logLevelUtils.ValidLogLevel(desiredOperatorLogLevel) {
		return reconcile.Result{}, fmt.Errorf("the operatorLogLevel defined for the cloudcredential object is invalid")
	}
```

Reply:
I have split both compound err checks to follow the recommended pattern.